### ISSUE TITLE
Suggest gas prices from ethgasstation.info

### ISSUE THREAD

**Issue by freeatnet** [forkdelta/forkdelta.github.io#2]
The interface should suggest an optimal gas price based on information from https://ethgasstation.info/.

**Comment by mariohm1311** [comment]
Actually, I'm pretty sure you could just make the default price change according to ethgasstation.info. That'd be a nice addition!

**Comment by freeatnet** [comment]
@mariohm1311 For sure, default should be adjusted automatically, but I wonder if we should also give users quick access to ethgasstation's "SafeLow" and "Fast" prices, too.

**Comment by mariohm1311** [comment]
Definitely, include those options too with a small explanation / median
time for each level.

On Jan 8, 2018 11:34 PM, "Arseniy Ivanov" <notifications@github.com> wrote:

@mariohm1311 <https://github.com/mariohm1311> For sure, default should be
adjusted automatically, but I wonder if we should also give users quick
access to ethgasstation's "SafeLow" and "Fast" prices, too.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/forkdelta/forkdelta.github.io/issues/2#issuecomment-356118403>,
or mute the thread

**Comment by wordsandstuff** [comment]
Please correct me if i am wrong, but the gas prices needs to be higher than the ethergas station price as the transactions are going to a contract witch require more gas.

**Comment by freeatnet** [comment]
@wordsandstuff Do you have a source for that information?

To clarify: "gas" is a measure of computation required to perform a transaction,. It is true that contracts require more "gas" because they require more computational steps than a regular fund transfer; however, a transaction requires the same amount of "gas" at any time. The sender can only set the maximum amount of computation they allow a transaction to take, "gas limit". Contracts typically require gas limit of 250,000.

You buy units of computation from miners at a price. That price is referred to as "gas price". This price can fluctuate a bit depending on network-wide demand for computation. This is what we're seeking to adjust dynamically.

Now: that is the theory of it. ethgasstation.info has a small note saying that gas prices may be different for addresses with a large number of pending transactions. @wordsandstuff, if you have any info on that, that'd be most welcome.

**Comment by freeatnet** [comment]
NB: MetaMask is experimenting with UI for letting the user set gas prices intuitively, too! https://github.com/MetaMask/metamask-extension/issues/3037

**Comment by mariohm1311** [comment]
Yup, you got it right. What you need to adjust on a per-transaction basis
is the gas limit, since you can already pull the gas price from
EthGasStation.

The quick workaround which has been commented is just leaving the gas limit
on 250,000 (the maximum), and have the network return the unused amount. I
would prefer it to adjust on a request basis instead, since if you are
making several transactions you could be incurring on 3-4 times the amount
of has you need, which might be more than the available ETH and therefore
you wouldn't be able to pay (even if most of that gas will not be used).
I'm not sure how straightforward it'd be to implement it, but Metamask
automatically sets up the gas limit for you, so I guess it isn't that
complicated.

**Comment by freeatnet** [comment]
> and have the network return the unused amount
That already happens. The transaction fee taken from your account is the lesser of (gas limit * gas price) and (gas usage * gas price).

You also typically cannot send a transaction until the previous one is mined, so multiple transactions shouldn't make a difference.

**Comment by mariohm1311** [comment]
I'm pretty sure I've sent transactions back to back by hoping they would be
verified in the right order and setting the nonce manually. I'm not quite
sure though.

**Comment by mhyytine** [comment]
What about making the default gas price equal to ethgasstation price plus one? Or would that just be overkill?

**Comment by mariohm1311** [comment]
That's what I usually do. Most of the times, adding one to the median price
nets a sizeable improvement (since the median price is used by many
services).

**Comment by banteg** [comment]
a simpler api for gas price: https://gasprice.poa.network/

i'm the author, see https://github.com/banteg/gasprice
